src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -12,16 +14,56 @@ namespace Microsoft.Extensions.Logging.Console
     [UnsupportedOSPlatform("browser")]
     internal class ConsoleLoggerProcessor : IDisposable
     {
-        private const int _maxQueuedMessages = 1024;
+        private static int s_processorCount;
+        private int _processorId;
+        private static List<ConsoleLoggerProcessor> s_processors = new List<ConsoleLoggerProcessor>();
```

Comment:
This needs to be a list of weak references or something needs to remove Processors from the list at some point, otherwise the List could grow without bound. Silly code example:
```
for(int i = 0; i < 10_000_000_000; i++) { new ConsoleProvider(); } // this loop shouldn't OOM
```

Reply:
> Why is this static?

Because metrics are process global state.

Reply:
removed s_processors